test/foundry/PermissionSplitter.t.sol

```diff
@@ -162,25 +180,62 @@ contract PermissionSplitterTest is Test {
         vm.stopPrank();
     }
 
-    function testFallback() public {
+    function testTransferOwnership() public {
+        vm.expectRevert(PROXY_NOT_ALLOWED_TO_CALL_ERROR);
+        hubPoolProxy.transferOwnership(defaultAdmin);
+
+        // Should be able to transfer ownership back to default admin in an emergency.
+        vm.startPrank(defaultAdmin);
+        hubPoolProxy.transferOwnership(defaultAdmin);
+        assertEq(hubPool.owner(), defaultAdmin);
+
+        hubPool.transferOwnership(address(permissionSplitter));
+        assertEq(hubPool.owner(), address(permissionSplitter));
+    }
+
+    /// forge-config: default.fuzz.runs = 300
+    function testFallback(bytes4 randomFuncSelector, bytes memory randomCalldata) public {
+        // random funcSelector doesn't collide with hub pool:
+        for (uint256 i = 0; i < hubPoolSelectors.length; i++) {
+            if (hubPoolSelectors[i] == randomFuncSelector) vm.assume(false);
+        }
+        // random funcSelector doesn't collide with proxy:
+        for (uint256 i = 0; i < proxySelectors.length; i++) {
+            if (proxySelectors[i] == randomFuncSelector) vm.assume(false);
+        }
+
         // Calling a function that doesn't exist on target or PermissionSplitter calls the HubPool's
         // fallback function which wraps any msg.value into wrapped native token.
         uint256 balBefore = address(hubPool).balance;
 
         // Calling fake function as admin with no value succeeds and does nothing.
         vm.prank(defaultAdmin);
-        (bool success1, ) = address(hubPoolProxy).call("doesNotExist()");
+        (bool success1, ) = address(hubPoolProxy).call(abi.encodeWithSelector(randomFuncSelector, randomCalldata));
         assertTrue(success1);
 
         // Calling fake function as admin with value also succeeds and wraps the msg.value
         // and then does nothing.
         vm.deal(defaultAdmin, 1 ether);
         vm.prank(defaultAdmin);
-        (bool success2, bytes memory reason) = address(hubPoolProxy).call{ value: 1 ether }("doesNotExist()");
+        (bool success2, ) = address(hubPoolProxy).call{ value: 1 ether }(
+            abi.encodeWithSelector(randomFuncSelector, randomCalldata)
+        );
         assertTrue(success2);
         assertEq(address(hubPool).balance, balBefore);
     }
 
+    function testReceive() public {
+        // Sending ETH to the proxy should trigger the target's receive() function, which on the HubPool wraps
+        // any msg.value into wrapped native token.
+        uint256 balBefore = address(hubPool).balance;
+
+        vm.deal(defaultAdmin, 1 ether);
+        vm.prank(defaultAdmin);
+        (bool success, ) = address(hubPoolProxy).call{ value: 1 ether }("");
+        assertTrue(success);
+        assertEq(address(hubPool).balance, balBefore);
```

Comment:
nit: should we also check the balance of the proxy to make sure the ETH got through?